pkg/machines/scripts/create_machine.sh

```diff
@@ -43,17 +43,17 @@ if [ "$OS" = "other-os" -o  -z "$OS" ]; then
     OS="auto"
 fi
 
-FIRST_1_SOURCE="`echo "$SOURCE" | cut -c 1`"
-FIRST_3_SOURCE="`echo "$SOURCE" | cut -c -3`"
-FIRST_4_SOURCE="`echo "$SOURCE" | cut -c -4`"
-
-if [ "$START_VM" = "true" -a "$FIRST_1_SOURCE" = "/" -a ! -f "$SOURCE" ]; then
+if [ "$START_VM" = "true" -a "${SOURCE#/}" != "$SOURCE" -a ! -f "$SOURCE" ]; then
     echo "$SOURCE does not exist or is not a file" 1>&2
     exit 1
 fi
 
-if [ "$START_VM" = "true" -a \( "$FIRST_1_SOURCE" = "/" -o "$FIRST_4_SOURCE" = "http" -o "$FIRST_3_SOURCE" = "ftp" -o "$FIRST_3_SOURCE" = "nfs" \) ]; then
-    LOCATION_PARAM="--cdrom $SOURCE"
+if [ "$START_VM" = "true" ]; then
+    if [ "${SOURCE#/}" != "$SOURCE" ] && [ "${SOURCE##*.}" = iso ]; then
```

Comment:
Do we need to have the `iso` check in place? Sure it would be nice to validate whether the image is ISO image but the file name doesn't have to end with `.iso`.  For example, it can be `.ISO` and that would fail to start installation because it would use `--location` instead.

Reply:
If I don't have the ```iso``` check we 'll have problem with local installation trees matched in the first branch of this condition. I also checked for the distros that we support and when you download the ISO the extensions of the file is always `.iso`.
If you think people often change the extensions in their ISO files let's add a more efficient check.

Reply:
Exactly :) usually people have `.iso` but it doesn't have to be the case and the same way, usually you have `iso` image and not an `os-tree`. In addition you can have an `iso` with different suffix.

How about we check whether the pat is directory or file, that's the main difference between some bootable image and os-tree.

Reply:
Ah and so I 'll remove and the check above.
https://github.com/cockpit-project/cockpit/blob/35348da7d60e3d0d0a910cedf18ac52e7f0170a2/pkg/machines/scripts/create_machine.sh#L50

Since we support directories as well with --location.

Reply:
Right, that can be removed.

Reply:
Actually there is check for non existent files/directories in front end, so the check inside the script can be safely removed.